ytview_2  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/

reating network "youtube-viewer-master_default" with the default driver
Creating youtube-viewer-master_ytview_1 ... done
Creating youtube-viewer-master_ytview_2 ... done
Creating youtube-viewer-master_ytview_3 ... done
Creating youtube-viewer-master_ytview_4 ... done
Creating youtube-viewer-master_ytview_5 ... done
Attaching to youtube-viewer-master_ytview_1, youtube-viewer-master_ytview_2, youtube-viewer-master_ytview_5, youtube-viewer-master_ytview_4, youtube-viewer-master_ytview_3
ytview_2  | [INFO] [11:45:48 PM] Preparing to generate 96 views. Target URL(s): https://www.youtube.com/watch?v=dQw4w9WgXcQ Duration: 50 seconds                                                                                                
ytview_2  | [INFO] [11:45:48 PM] App running in production. Will use rotating proxy via TOR.
ytview_2  | [INFO] [11:45:48 PM] -------- Writing Tor Config -------- 
ytview_2  | 
ytview_2  | [SUCCESS] [11:45:48 PM] PORT 9057 written in tor config
ytview_2  | [SUCCESS] [11:45:48 PM] PORT 9056 written in tor config
ytview_2  | [SUCCESS] [11:45:48 PM] PORT 9055 written in tor config
ytview_2  | [SUCCESS] [11:45:48 PM] PORT 9054 written in tor config
ytview_2  | [SUCCESS] [11:45:48 PM] PORT 9053 written in tor config
ytview_2  | [SUCCESS] [11:45:48 PM] PORT 9052 written in tor config
ytview_2  | [INFO] [11:45:48 PM] Starting TOR.
ytview_2  | [INFO] [11:45:48 PM] Failed to stop TOR. Usually this is a no op but ensure the subsequent attempts are using different IPs.                                                                                                        
ytview_2  | [SUCCESS] [11:45:48 PM] Started TOR successfully
ytview_1  | [INFO] [11:45:48 PM] Preparing to generate 96 views. Target URL(s): https://www.youtube.com/watch?v=dQw4w9WgXcQ Duration: 50 seconds                                                                                                
ytview_1  | [INFO] [11:45:48 PM] App running in production. Will use rotating proxy via TOR.
ytview_1  | [INFO] [11:45:48 PM] -------- Writing Tor Config -------- 
ytview_1  | 
ytview_1  | [SUCCESS] [11:45:48 PM] PORT 9056 written in tor config
ytview_1  | [SUCCESS] [11:45:48 PM] PORT 9055 written in tor config
ytview_1  | [SUCCESS] [11:45:48 PM] PORT 9054 written in tor config
ytview_1  | [SUCCESS] [11:45:48 PM] PORT 9053 written in tor config
ytview_1  | [SUCCESS] [11:45:48 PM] PORT 9052 written in tor config
ytview_1  | [SUCCESS] [11:45:48 PM] PORT 9057 written in tor config
ytview_1  | [INFO] [11:45:48 PM] Starting TOR.
ytview_1  | [INFO] [11:45:48 PM] Failed to stop TOR. Usually this is a no op but ensure the subsequent attempts are using different IPs.                                                                                                        
ytview_1  | [SUCCESS] [11:45:48 PM] Started TOR successfully
ytview_4  | [INFO] [11:45:48 PM] Preparing to generate 96 views. Target URL(s): https://www.youtube.com/watch?v=dQw4w9WgXcQ Duration: 50 seconds                                                                                                
ytview_4  | [INFO] [11:45:48 PM] App running in production. Will use rotating proxy via TOR.
ytview_4  | [INFO] [11:45:48 PM] -------- Writing Tor Config -------- 
ytview_4  | 
ytview_4  | [SUCCESS] [11:45:48 PM] PORT 9057 written in tor config
ytview_4  | [SUCCESS] [11:45:48 PM] PORT 9056 written in tor config
ytview_4  | [SUCCESS] [11:45:48 PM] PORT 9055 written in tor config
ytview_4  | [SUCCESS] [11:45:48 PM] PORT 9053 written in tor config
ytview_4  | [SUCCESS] [11:45:48 PM] PORT 9052 written in tor config
ytview_4  | [SUCCESS] [11:45:48 PM] PORT 9054 written in tor config
ytview_4  | [INFO] [11:45:48 PM] Starting TOR.
ytview_4  | [INFO] [11:45:48 PM] Failed to stop TOR. Usually this is a no op but ensure the subsequent attempts are using different IPs.                                                                                                        
ytview_4  | [SUCCESS] [11:45:48 PM] Started TOR successfully
ytview_5  | [INFO] [11:45:48 PM] Preparing to generate 96 views. Target URL(s): https://www.youtube.com/watch?v=dQw4w9WgXcQ Duration: 50 seconds                                                                                                
ytview_5  | [INFO] [11:45:48 PM] App running in production. Will use rotating proxy via TOR.
ytview_5  | [INFO] [11:45:48 PM] -------- Writing Tor Config -------- 
ytview_5  | 
ytview_5  | [SUCCESS] [11:45:48 PM] PORT 9057 written in tor config
ytview_5  | [SUCCESS] [11:45:48 PM] PORT 9056 written in tor config
ytview_5  | [SUCCESS] [11:45:48 PM] PORT 9055 written in tor config
ytview_5  | [SUCCESS] [11:45:48 PM] PORT 9054 written in tor config
ytview_5  | [SUCCESS] [11:45:48 PM] PORT 9053 written in tor config
ytview_5  | [SUCCESS] [11:45:48 PM] PORT 9052 written in tor config
ytview_5  | [INFO] [11:45:48 PM] Starting TOR.
ytview_5  | [INFO] [11:45:48 PM] Failed to stop TOR. Usually this is a no op but ensure the subsequent attempts are using different IPs.                                                                                                        
ytview_5  | [SUCCESS] [11:45:48 PM] Started TOR successfully
ytview_3  | [INFO] [11:45:49 PM] Preparing to generate 96 views. Target URL(s): https://www.youtube.com/watch?v=dQw4w9WgXcQ Duration: 50 seconds                                                                                                
ytview_3  | [INFO] [11:45:49 PM] App running in production. Will use rotating proxy via TOR.
ytview_3  | [INFO] [11:45:49 PM] -------- Writing Tor Config -------- 
ytview_3  | 
ytview_3  | [SUCCESS] [11:45:49 PM] PORT 9056 written in tor config
ytview_3  | [SUCCESS] [11:45:49 PM] PORT 9055 written in tor config
ytview_3  | [SUCCESS] [11:45:49 PM] PORT 9054 written in tor config
ytview_3  | [SUCCESS] [11:45:49 PM] PORT 9053 written in tor config
ytview_3  | [SUCCESS] [11:45:49 PM] PORT 9052 written in tor config
ytview_3  | [SUCCESS] [11:45:49 PM] PORT 9057 written in tor config
ytview_3  | [INFO] [11:45:49 PM] Starting TOR.
ytview_3  | [INFO] [11:45:49 PM] Failed to stop TOR. Usually this is a no op but ensure the subsequent attempts are using different IPs.                                                                                                        
ytview_3  | [SUCCESS] [11:45:50 PM] Started TOR successfully
ytview_1  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_2  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_1  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_5  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_4  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_2  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_1  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_5  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_1  | [ERROR] [11:46:23 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_4  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_1  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_1  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_5  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_1  | [INFO] [11:46:24 PM] View 1 - fulfilled
ytview_1  | [INFO] [11:46:24 PM] Success - 1     Failed - 0      Total - 1
ytview_1  | [INFO] [11:46:24 PM] View 2 - fulfilled
ytview_1  | [INFO] [11:46:24 PM] Success - 2     Failed - 0      Total - 2
ytview_1  | [INFO] [11:46:24 PM] View 3 - fulfilled
ytview_1  | [INFO] [11:46:24 PM] Success - 3     Failed - 0      Total - 3
ytview_1  | [INFO] [11:46:24 PM] View 4 - fulfilled
ytview_1  | [INFO] [11:46:24 PM] Success - 4     Failed - 0      Total - 4
ytview_1  | [INFO] [11:46:24 PM] View 5 - fulfilled
ytview_1  | [INFO] [11:46:24 PM] Success - 5     Failed - 0      Total - 5
ytview_1  | [INFO] [11:46:24 PM] View 6 - fulfilled
ytview_1  | [INFO] [11:46:24 PM] Success - 6     Failed - 0      Total - 6
ytview_1  | [INFO] [11:46:24 PM] Starting TOR.
ytview_2  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_2  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_5  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_2  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_4  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_2  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_4  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_5  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_5  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_4  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_4  | [ERROR] [11:46:24 PM] Entire view action in a batch failed. Error: Error: net::ERR_TIMED_OUT at https://api.ipify.org/                                                                                                              
ytview_5  | [INFO] [11:46:24 PM] View 1 - fulfilled
ytview_5  | [INFO] [11:46:24 PM] Success - 1     Failed - 0      Total - 1
ytview_5  | [INFO] [11:46:24 PM] View 2 - fulfilled
ytview_5  | [INFO] [11:46:24 PM] Success - 2     Failed - 0      Total - 2
ytview_5  | [INFO] [11:46:24 PM] View 3 - fulfilled
ytview_5  | [INFO] [11:46:24 PM] Success - 3     Failed - 0      Total - 3
ytview_5  | [INFO] [11:46:24 PM] View 4 - fulfilled
ytview_5  | [INFO] [11:46:24 PM] Success - 4     Failed - 0      Total - 4
ytview_5  | [INFO] [11:46:24 PM] View 5 - fulfilled
ytview_5  | [INFO] [11:46:24 PM] Success - 5     Failed - 0      Total - 5
ytview_5  | [INFO] [11:46:24 PM] View 6 - fulfilled
ytview_5  | [INFO] [11:46:24 PM] Success - 6     Failed - 0      Total - 6
ytview_2  | [INFO] [11:46:24 PM] View 1 - fulfilled
ytview_2  | [INFO] [11:46:24 PM] Success - 1     Failed - 0      Total - 1
ytview_2  | [INFO] [11:46:24 PM] View 2 - fulfilled
ytview_5  | [INFO] [11:46:24 PM] Starting TOR.
ytview_4  | [INFO] [11:46:24 PM] View 1 - fulfilled
ytview_4  | [INFO] [11:46:24 PM] Success - 1     Failed - 0      Total - 1
ytview_2  | [INFO] [11:46:24 PM] Success - 2     Failed - 0      Total - 2
ytview_2  | [INFO] [11:46:24 PM] View 3 - fulfilled
ytview_2  | [INFO] [11:46:24 PM] Success - 3     Failed - 0      Total - 3
ytview_4  | [INFO] [11:46:24 PM] View 2 - fulfilled
ytview_4  | [INFO] [11:46:24 PM] Success - 2     Failed - 0      Total - 2
ytview_2  | [INFO] [11:46:24 PM] View 4 - fulfilled
ytview_2  | [INFO] [11:46:24 PM] Success - 4     Failed - 0      Total - 4
ytview_2  | [INFO] [11:46:24 PM] View 5 - fulfilled
ytview_2  | [INFO] [11:46:24 PM] Success - 5     Failed - 0      Total - 5
ytview_4  | [INFO] [11:46:24 PM] View 3 - fulfilled
ytview_4  | [INFO] [11:46:24 PM] Success - 3     Failed - 0      Total - 3
ytview_4  | [INFO] [11:46:24 PM] View 4 - fulfilled
ytview_2  | [INFO] [11:46:24 PM] View 6 - fulfilled
ytview_2  | [INFO] [11:46:24 PM] Success - 6     Failed - 0      Total - 6
ytview_2  | [INFO] [11:46:24 PM] Starting TOR.
ytview_4  | [INFO] [11:46:24 PM] Success - 4     Failed - 0      Total - 4
ytview_4  | [INFO] [11:46:24 PM] View 5 - fulfilled
ytview_4  | [INFO] [11:46:24 PM] Success - 5     Failed - 0      Total - 5
ytview_4  | [INFO] [11:46:24 PM] View 6 - fulfilled
ytview_4  | [INFO] [11:46:24 PM] Success - 6     Failed - 0      Total - 6
ytview_4  | [INFO] [11:46:24 PM] Starting TOR.
ytview_1  | [SUCCESS] [11:46:24 PM] Started TOR successfully
ytview_4  | [SUCCESS] [11:46:24 PM] Started TOR successfully
ytview_5  | [SUCCESS] [11:46:24 PM] Started TOR successfully
ytview_2  | [SUCCESS] [11:46:24 PM] Started TOR successfully
ytview_3  | [ERROR] [11:46:48 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 51.68.201.118 was probably blocked.
ytview_3  | [ERROR] [11:46:49 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.193 was probably blocked.
ytview_4  | [ERROR] [11:47:04 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.152 was probably blocked.
ytview_4  | [ERROR] [11:47:10 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 199.249.230.75 was probably blocked.
ytview_4  | [ERROR] [11:47:11 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.201 was probably blocked.
ytview_5  | [ERROR] [11:47:12 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.214 was probably blocked.
ytview_5  | [ERROR] [11:47:13 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.130 was probably blocked.
ytview_1  | [ERROR] [11:47:13 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.135 was probably blocked.        
ytview_5  | [ERROR] [11:47:13 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 79.137.79.167 was probably blocked.                     
ytview_3  | [ERROR] [11:47:17 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 109.70.100.34 was probably blocked.                     
ytview_5  | [ERROR] [11:47:17 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.136 was probably blocked.                             
ytview_1  | [ERROR] [11:47:17 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.14 was probably blocked.                                            
ytview_2  | [ERROR] [11:47:20 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.10 was probably blocked.                                            
ytview_2  | [ERROR] [11:47:20 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.141 was probably blocked.                                                 
ytview_2  | [ERROR] [11:47:20 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 51.89.226.199 was probably blocked.                                                   
ytview_2  | [ERROR] [11:47:20 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.101.18 was probably blocked.
ytview_1  | [ERROR] [11:47:21 PM] An attempt to view https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 51.89.114.178 was probably blocked.
ytview_5  | [SUCCESS] [11:48:10 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.102.6 for 47.45049369376864 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)                                                                                                                                                                      
ytview_3  | [SUCCESS] [11:48:19 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 109.70.100.23 for 55.213079007762886 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)                                                                                                                                                                     
ytview_3  | [SUCCESS] [11:48:22 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.102.8 for 56.25011003268105 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)                                                                                                                                                                      
ytview_3  | [SUCCESS] [11:48:23 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 185.220.100.254 for 52.811363550802646 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)                                                                                                                                                                   
ytview_3  | [INFO] [11:48:23 PM] View 1 - fulfilled
ytview_3  | [INFO] [11:48:23 PM] Success - 1     Failed - 0      Total - 1
ytview_3  | [INFO] [11:48:23 PM] View 2 - fulfilled
ytview_3  | [INFO] [11:48:23 PM] Success - 2     Failed - 0      Total - 2
ytview_3  | [INFO] [11:48:23 PM] View 3 - fulfilled
ytview_3  | [INFO] [11:48:23 PM] Success - 3     Failed - 0      Total - 3
ytview_3  | [INFO] [11:48:23 PM] View 4 - fulfilled
ytview_3  | [INFO] [11:48:23 PM] Success - 4     Failed - 0      Total - 4
ytview_3  | [INFO] [11:48:23 PM] View 5 - fulfilled
ytview_3  | [INFO] [11:48:23 PM] Success - 5     Failed - 0      Total - 5
ytview_3  | [INFO] [11:48:23 PM] View 6 - fulfilled
ytview_3  | [INFO] [11:48:23 PM] Success - 6     Failed - 0      Total - 6
ytview_3  | [INFO] [11:48:23 PM] Starting TOR.
ytview_3  | [SUCCESS] [11:48:23 PM] Started TOR successfully
ytview_5  | [SUCCESS] [11:48:24 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 46.165.245.154 for 51.27143703589482 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)                                                                                                                                                                     
ytview_5  | [INFO] [11:48:24 PM] View 7 - fulfilled
ytview_5  | [INFO] [11:48:24 PM] Success - 7     Failed - 0      Total - 7
ytview_5  | [INFO] [11:48:24 PM] View 8 - fulfilled
ytview_5  | [INFO] [11:48:24 PM] Success - 8     Failed - 0      Total - 8
ytview_5  | [INFO] [11:48:24 PM] View 9 - fulfilled
ytview_5  | [INFO] [11:48:24 PM] Success - 9     Failed - 0      Total - 9
ytview_5  | [INFO] [11:48:24 PM] View 10 - fulfilled
ytview_5  | [INFO] [11:48:24 PM] Success - 10    Failed - 0      Total - 10
ytview_5  | [INFO] [11:48:24 PM] View 11 - fulfilled
ytview_5  | [INFO] [11:48:24 PM] Success - 11    Failed - 0      Total - 11
ytview_5  | [INFO] [11:48:24 PM] View 12 - fulfilled
ytview_5  | [INFO] [11:48:24 PM] Success - 12    Failed - 0      Total - 12
ytview_5  | [INFO] [11:48:24 PM] Starting TOR.
ytview_5  | [SUCCESS] [11:48:24 PM] Started TOR successfully
ytview_2  | [SUCCESS] [11:48:27 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 179.43.167.230 for 57.15203642770241 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)
ytview_2  | [SUCCESS] [11:48:28 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 192.42.116.16 for 57.02661892709723 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)
ytview_2  | [INFO] [11:48:28 PM] View 7 - fulfilled
ytview_2  | [INFO] [11:48:28 PM] Success - 7     Failed - 0      Total - 7
ytview_2  | [INFO] [11:48:28 PM] View 8 - fulfilled
ytview_2  | [INFO] [11:48:28 PM] Success - 8     Failed - 0      Total - 8
ytview_2  | [INFO] [11:48:28 PM] View 9 - fulfilled
ytview_2  | [INFO] [11:48:28 PM] Success - 9     Failed - 0      Total - 9
ytview_2  | [INFO] [11:48:28 PM] View 10 - fulfilled
ytview_2  | [INFO] [11:48:28 PM] Success - 10    Failed - 0      Total - 10
ytview_2  | [INFO] [11:48:28 PM] View 11 - fulfilled
ytview_2  | [INFO] [11:48:28 PM] Success - 11    Failed - 0      Total - 11
ytview_2  | [INFO] [11:48:28 PM] View 12 - fulfilled
ytview_2  | [INFO] [11:48:28 PM] Success - 12    Failed - 0      Total - 12
ytview_2  | [INFO] [11:48:28 PM] Starting TOR.
ytview_2  | [SUCCESS] [11:48:28 PM] Started TOR successfully
ytview_4  | [SUCCESS] [11:48:31 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 195.176.3.23 for 45.47183420704917 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)
ytview_1  | [SUCCESS] [11:48:36 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 45.9.148.217 for 51.800268099981956 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)
ytview_1  | [SUCCESS] [11:48:38 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 87.118.92.43 for 58.13667194533737 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)
ytview_4  | [SUCCESS] [11:48:40 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 109.70.100.35 for 55.814975266193755 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)
ytview_1  | [SUCCESS] [11:48:42 PM] Attempted https://www.youtube.com/watch?v=dQw4w9WgXcQ with IP: 77.247.181.162 for 56.57729011613876 seconds. (Init View Count: 356 Current View Count: 356 Views added this session: 0)
ytview_1  | [INFO] [11:48:42 PM] View 7 - fulfilled
ytview_1  | [INFO] [11:48:42 PM] Success - 7     Failed - 0      Total - 7
ytview_1  | [INFO] [11:48:42 PM] View 8 - fulfilled
ytview_1  | [INFO] [11:48:42 PM] Success - 8     Failed - 0      Total - 8
ytview_1  | [INFO] [11:48:42 PM] View 9 - fulfilled
ytview_1  | [INFO] [11:48:42 PM] Success - 9     Failed - 0      Total - 9
ytview_1  | [INFO] [11:48:42 PM] View 10 - fulfilled
ytview_1  | [INFO] [11:48:42 PM] Success - 10    Failed - 0      Total - 10
ytview_1  | [INFO] [11:48:42 PM] View 11 - fulfilled
ytview_1  | [INFO] [11:48:42 PM] Success - 11    Failed - 0      Total - 11
ytview_1  | [INFO] [11:48:42 PM] View 12 - fulfilled
ytview_1  | [INFO] [11:48:42 PM] Success - 12    Failed - 0      Total - 12
ytview_1  | [INFO] [11:48:42 PM] Starting TOR.
ytview_1  | [SUCCESS] [11:48:42 PM] Started TOR successfully

--- a/README.md
+++ b/README.md
@@ -44,7 +44,7 @@ The following commands will help create a docker image, build the app and run it
  - **View Action**: This represents a single browser instance picking up a fixed number of urls from the pool and visiting them sequentially.
  - **View Action Count**: A single browsing session will watch these many videos sequentially.
  - **Total Count** - Total number of view actions. Ensure this number is exactly divisible by **Batch Count** for optimal resource utilisation.
- - **View Duration** - Max duration in seconds of a single view in view action. Actual view duration will be +/- 16.6% of this number.
+ - **View Duration** - Average duration in seconds of a single view in view action. Actual view duration will be +/- 16.6% of this number.
 
 You may choose to alter the above params in `utils/constants/index.js` for fine tuning according to your needs. 
 

--- a/handlers/startViewing.handler.js
+++ b/handlers/startViewing.handler.js
@@ -15,14 +15,17 @@ const startViewingHandler = async (options, index) => {
     const port = options.startPort + i;
     promiseArr.push(YTBrowserService.viewVideosInBatch({ ...options, port }));
   }
-  return Promise.allSettled(promiseArr).then((settedPromises) => _each(settedPromises, ({ status }, i) => {
-    total += 1;
-    if (status === 'fulfilled') successes += 1;
-    else failures += 1;
+  return Promise.allSettled(promiseArr).then((settedPromises) => {
+    logger.info('Batch Summary -');
+    _each(settedPromises, ({ status }, i) => {
+      total += 1;
+      if (status === 'fulfilled') successes += 1;
+      else failures += 1;
 
-    logger.info(`View ${index * options.batchCount + i + 1} - ${status}`);
-    logger.info(`Success - ${successes}\t Failed - ${failures}\t Total - ${total}`);
-  }));
+      logger.info(`View ${index * options.batchCount + i + 1} - ${status}`);
+      logger.info(`Fulfilled - ${successes}\t Failed - ${failures}\t Total - ${total}`);
+    });
+  });
 };
 
 module.exports = startViewingHandler;

--- a/services/tor.service.js
+++ b/services/tor.service.js
@@ -6,7 +6,7 @@ const { IS_PROD } = require('../utils/constants');
 const writeTorConfig = async (startPort, count) => {
   if (!IS_PROD) return Promise.resolve();
   logger.info('App running in production. Will use rotating proxy via TOR.');
-  logger.info('-------- Writing Tor Config -------- \n');
+  logger.info(' Writing Tor Config');
   await execWithPromise('touch /etc/tor/torrc && echo > /etc/tor/torrc');
   const promiseArr = [];
   for (let i = 0; i < count; i += 1) {
@@ -14,11 +14,14 @@ const writeTorConfig = async (startPort, count) => {
     promiseArr.push(
       execWithPromise(
         `echo "SocksPort ${port}" >> /etc/tor/torrc`,
-      ).then(() => logger.success(`PORT ${port} written in tor config`)),
+      ).then(() => logger.debug(`PORT ${port} written in tor config`)),
     );
   }
-  return Promise.all(promiseArr).catch((error) => {
-    logger.error(`One or more ports couldn't be written in tor config. Error: ${error}`);
+  return Promise.all(promiseArr).then(() => {
+    logger.success('Tor Config written successfully.');
+  }).catch((error) => {
+    logger.error('One or more ports couldn\'t be written into tor config.');
+    logger.debug(error);
     throw new Error();
   });
 };
@@ -27,8 +30,9 @@ const stopTor = async () => {
   if (!IS_PROD) return;
   try {
     await execWithPromise('pkill -9 -f "tor"');
-  } catch {
-    logger.info('Failed to stop TOR. Usually this is a no op but ensure the subsequent attempts are using different IPs.');
+  } catch (error) {
+    logger.warn('Failed to stop TOR. Usually this is a no op but ensure the subsequent attempts are using different IPs.');
+    logger.debug(error);
   }
 };
 
@@ -39,8 +43,9 @@ const startTor = async () => {
   try {
     await execWithPromise('/usr/bin/tor --RunAsDaemon 1');
     logger.success('Started TOR successfully');
-  } catch {
+  } catch (error) {
     logger.error('Failed to start TOR.');
+    logger.debug(error);
     throw new Error();
   }
 };

--- a/services/youtubeBrowser.service.js
+++ b/services/youtubeBrowser.service.js
@@ -27,7 +27,8 @@ const viewVideosInBatch = async ({ targetUrls, durationInSeconds, port }) => {
     await watchVideosInSequence(page, ipAddr, targetUrlsForAction, durationInSeconds);
     await page.close();
   } catch (error) {
-    logger.error(`Entire view action in a batch failed. Error: ${error}`);
+    logger.warn('Entire view action in a batch failed. Waiting for TOR to acquire a new set of IPs');
+    logger.debug(error);
   } finally {
     await browser.close();
   }

--- a/utils/logger.js
+++ b/utils/logger.js
@@ -1,13 +1,20 @@
 const chalk = require('chalk');
 
+const { IS_PROD } = require('./constants');
+
 const store = {};
 
-const info = (message) => console.log(`${chalk.blue.inverse(`[INFO] [${(new Date()).toLocaleTimeString()}]`)} ${chalk.blue(message)}`);
-const error = (message) => console.log(`${chalk.red.inverse(`[ERROR] [${(new Date()).toLocaleTimeString()}]`)} ${chalk.red(message)}`);
-const success = (message) => console.log(`${chalk.green.inverse(`[SUCCESS] [${(new Date()).toLocaleTimeString()}]`)} ${chalk.green(message)}`);
+const info = (message) => console.log(`${chalk.white.inverse(` [${(new Date()).toLocaleTimeString()}] - INFO    `)} ${chalk.white(message)}`);
+const error = (message) => console.log(`${chalk.red.inverse(` [${(new Date()).toLocaleTimeString()}] - ERROR   `)} ${chalk.red(message)}`);
+const success = (message) => console.log(`${chalk.green.inverse(` [${(new Date()).toLocaleTimeString()}] - SUCCESS `)} ${chalk.green(message)}`);
+const debug = (message) => {
+  if (IS_PROD) return;
+  console.log(`${chalk.magenta.inverse(` [${(new Date()).toLocaleTimeString()}] - DEBUG   `)} ${chalk.magenta(message)}`);
+};
+const warn = (message) => console.log(`${chalk.yellow.inverse(` [${(new Date()).toLocaleTimeString()}] - WARN    `)} ${chalk.yellow(message)}`);
 
 const logFailedAttempt = (url, ipAddr) => {
-  error(`An attempt to view ${url} with IP: ${ipAddr} was probably blocked.`);
+  warn(`An attempt to view ${url} with IP: ${ipAddr} was probably blocked.`);
 };
 
 const logCount = async (page, url, ipAddr, duration) => {
@@ -27,5 +34,7 @@ module.exports = {
   logFailedAttempt,
   info,
   error,
+  warn,
   success,
+  debug,
 };

--- a/utils/urlReader.js
+++ b/utils/urlReader.js
@@ -14,7 +14,8 @@ const urlReader = (filename) => {
     if (_isEmpty(urlArray)) throw new Error(`No urls found. Make sure ${filename} contains at least one url to proceed`);
     return urlArray;
   } catch (error) {
-    logger.error(`An error occurred while reading urls. Error: ${error}`);
+    logger.error('An error occurred while reading urls');
+    logger.debug(error);
     throw new Error();
   }
 };

